src/components/svelte-web.ts

```diff
@@ -254,16 +254,20 @@ export default function registerWebComponent(
 
     addEventListener(
       event: string,
-      callback: Callback,
+      eventHandler: EventListenerOrEventListenerObject,
       options?: boolean | AddEventListenerOptions
     ) {
       const svelteEvent = events[event]
       if (svelteEvent) {
+        const callback =
+          'handleEvent' in eventHandler
+            ? eventHandler.handleEvent.bind(eventHandler)
+            : eventHandler
```

Comment:
Should we being checking types here instead? Something like:

```js
if (typeof eventHandler === 'function') {
  callback = eventHandler
} else if ('handleEvent' in eventHandler) {
  callback = eventHandler.handleEvent.bind(eventHandler)
} else {
  throw Error("Must pass a function or include one in object as 'handleEvent'")
}
```

Reply:
tbh, I'm not too fussed either way - this is pretty niche. Happy to change it, but I kinda like it as is.

Reply:
It's definitely more succinct as is. Let's go for it!